### DISCUSSION TITLE
isort disagrees with pylint

### DISCUSSION THREAD

**Discussion by dfrankow** [PyCQA/isort#2015] [Q&A]
I put both isort (5.10.1) and pylint (2.13.4) into my pre-commit script. They disagree:

```
pylint...................................................................Failed
- hook id: pylint
- exit code: 16

************* Module audio
src/some_file.py:6:0: C0411: third party import "from gtts import gTTS" should be placed before "from a_file import AFile" (wrong-import-order)
```

When I move the third-party gtts above a_file, isort puts it back and annoys pylint.

So, sadly, I've disabled isort.

Best wishes.

**Comment by staticdev** [comment]
That is why I use isort + flake8, they work better together and I think flake8 is a bit superior these days IMHO. In any case I don't consider this an issue so I moved it to discussions.

You can see an example python library I use isort working together with a ton of linters: https://github.com/staticdev/human-readable/

**Comment by dfrankow** [comment]
I'm actually using both flake8 and pylint. They don't seem to disagree with each other. pylint just seems a lot pickier.

There are really two issues here:

1. isort disagrees with pylint
2. isort doesn't put third party imports before local code imports.

Sounds like you don't care about 1. What about 2?

If you don't care about that either, that is your prerogative. But, I probably won't use isort until it's okay with 1 and 2. Or, maybe I'll drop pylint and keep flake8, we'll see.

Best wishes.

**Comment by sztamas** [comment]
Hi there,

You could solve that disagreement by configuring isort to consider `a_file` as a first party import instead of a third party import.
Without knowing much about your project setup, these would be some pointers that might help you in doing that.

Do you have `src-paths` in your config?

https://pycqa.github.io/isort/docs/configuration/options.html#src-paths

Or if you need more flexibility you also have the`known_first_party` config option.

https://pycqa.github.io/isort/docs/configuration/options.html#known-first-party

I hope this helps,
Tamas

**Reply by dfrankow** [reply]
Oh, thank you!

**Reply by dfrankow** [reply]
I do wonder how with the current setup (no config) pylint knows but isort does not. Still this is helpful.

**Reply by sztamas** [reply]
I'm not sure without knowing how your project is set up and how and where from you are running `isort`.
I think isort used to do more guessing on where your own source files are, but that was causing all kind of issues and the guessing logic has been removed.
Maybe pylint has some of that "magic" guessing code that isort doesn't have anymore...

**Reply by dfrankow** [reply]
I did not fully understand src-paths. I had misconfigured it. I configured it correctly, and isort works fine. I think relying on explicit config is better than magic. Thank you!
